components/cart/cart.controller.js

```diff
@@ -1,3 +1,18 @@
-const cartController = function () {
+const cartController = function ($scope, cartService) {
+  this.finalCart = [];
+  this.totalPrice = 0;
+
+  this.$onInit = () => {
+    this.finalCart = cartService.cart;
```

Comment:
naming: this should be "cart", not "finalCart".

also, we should use a getter to get the cart, something like cartService.getCart()

Reply:
Renamed finalCart into cart.

I will update the cartService using an observable